library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2839,7 +2839,7 @@ impl<T> From<VecDeque<T>> for Vec<T> {
             let len = other.len();
             let cap = other.cap();
 
-            if other.head != 0 {
+            if other.tail != 0 {
```

Comment:
this bug was previously unreachable because `is_contiguous` returned false if the `head` was `0`.

uff